cmd/opampsupervisor/e2e_test.go

```diff
@@ -1261,6 +1261,129 @@ func TestSupervisorWritesAgentFilesToStorageDir(t *testing.T) {
 	require.FileExists(t, filepath.Join(storageDir, "effective.yaml"))
 }
 
+func TestSupervisorRemoteConfigApplyStatus(t *testing.T) {
```

Comment:
Can we get the time this test takes down? It takes 14s on my machine. Ideally it would take less than a second, but if we could get it down towards that it would be nice.

Reply:
It takes about the same time on the machine as well. Let me check what can be done here to bring it down.

Reply:
Now it is around with 7s. Half of that time is spent waiting for 10 health check attempts to fail. Should we make it configurable? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/af21ce7ff8834d6089f288a52e88a928e7ddad60/cmd/opampsupervisor/supervisor/supervisor.go#L1029

Reply:
Trying to make this configurable introduced one more option and made it hard to reason about configuration. I am going to do some refactoring and push changes to get feedback.

Reply:
Updated the supervisor to use the same config options.